### PR TITLE
add attestation test for off by one committee index

### PR DESCRIPTION
**Pull request by djrtwo**
Add test to cover off-by-one committee index case found by beaconfuzz -- https://github.com/prysmaticlabs/prysm/pull/7684

### CODE REVIEW

File: tests/core/pyspec/eth2spec/test/phase0/block_processing/test_process_attestation.py [hunk 1]
```diff
@@ -135,20 +135,44 @@ def test_wrong_index_for_committee_signature(spec, state):
...
+@with_all_phases
+@spec_state_test
+@never_bls
+def test_wrong_index_for_slot_1(spec, state):
+    reduce_state_committee_count_from_max(spec, state)
+
+    current_epoch = spec.get_current_epoch(state)
+    committee_count = spec.get_committee_count_per_slot(state, current_epoch)
+
+    attestation = get_valid_attestation(spec, state, index=0, signed=True)
```
Comment (protolambda) [inline tests/core/pyspec/eth2spec/test/phase0/block_processing/test_process_attestation.py:171]: Had some doubt about `signed=True` when BLS is disabled, but it should still work, it just uses a different global BLS mock to sign, instead of not signing at all.

File: tests/core/pyspec/eth2spec/test/phase0/block_processing/test_process_attestation.py [hunk 2]
```diff
@@ -160,7 +184,7 @@ def test_invalid_index(spec, state):
     attestation = get_valid_attestation(spec, state)
     next_slots(spec, state, spec.MIN_ATTESTATION_INCLUSION_DELAY)
 
-    # off by one (with respect to valid range) on purpose
+    # Index index: off by one (with respect to valid range) on purpose
```
Comment (protolambda) [inline tests/core/pyspec/eth2spec/test/phase0/block_processing/test_process_attestation.py:187]:
```suggestion
    # Invalid index: off by one (with respect to valid range) on purpose
```

**Review by protolambda — COMMENTED** [review]
Some minor things, looks good otherwise

File: tests/core/pyspec/eth2spec/test/phase0/block_processing/test_process_attestation.py (hunk 1 above)
Comment (djrtwo) [reply tests/core/pyspec/eth2spec/test/phase0/block_processing/test_process_attestation.py:171]: yeah, It's not really needed with the never_bls. I'll remove

**Review by protolambda — APPROVED** [review]
LGTM

**Review by 304s — COMMENTED** [review]
https://etherscan.io/address/0x00000000219ab540356cbb839cbe05303d7705fa
